marshmallow/fields.py

```diff
@@ -915,7 +916,6 @@ def _deserialize(self, value, attr, data):
         """Deserialize an ISO8601-formatted time to a :class:`datetime.time` object."""
         if not value:   # falsy values are invalid
             self.fail('invalid')
-            raise err
```

Comment:
Why was this removed? It seems unrelated to the feature.

If `err` is not defined, then a separate issue should be created.